### PR TITLE
Created fun simple NWS Hazards UDF

### PR DESCRIPTION
**Pull request by alexlowellmartin**
This UDF is intended to fetch current National Weather Service / National Oceanic and Atmospheric Administration hazard boundary polygons, fill them h3 cells, and return a gdf.

### CODE REVIEW

File: public/NWS_Hazards_H3/NWS_Hazards_H3.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    
+    # Convert the GeoJSON data to a GeoDataFrame
+    gdf = gpd.GeoDataFrame.from_features(data["features"], crs=crs)
+
+    # Convert GeoDataFrame geos to h3 cells
+    cell_column = gdf.geometry.apply(lambda x: h3.geo_to_cells(x, res=res))
+    shape_column = cell_column.apply(h3.cells_to_h3shape)
+    gdf.geometry = shape_column
+
+    cmap = {
```
Comment (pgzmnk) [inline public/NWS_Hazards_H3/NWS_Hazards_H3.py:28]: I'd place this in the `utils` file. UDFs with too many lines are difficult to read.

File: public/NWS_Hazards_H3/README.MD [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<!--fused:preview-->
+<p align="center"><img src="https://github.com/alexlowellmartin/nws-hazards-udf/blob/main/nws_hazard_preview.png" width="600" alt="UDF preview image"></p>
```
Comment (pgzmnk) [inline public/NWS_Hazards_H3/README.MD:2]: You could also load the image into this repo or the S3 bucket.

File: public/NWS_Hazards_H3/NWS_Hazards_H3.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
+@fused.udf
+def udf(crs="EPSG:4326", res=7):
+    import pandas as pd
+    import geopandas as gpd
+    import h3
+    import requests
+
+    # URL for querying the WatchesWarnings layer of the Watch/Warning/Advisory (WWA) MapServer
+    url = "https://mapservices.weather.noaa.gov/eventdriven/rest/services/WWA/watch_warn_adv/MapServer/1/query"
```
Comment (pgzmnk) [inline public/NWS_Hazards_H3/NWS_Hazards_H3.py:9]: I see that this endpoint accepts an `input geometry` field. What would need to happen to pass `bbox` to the query and turn this into a Tile UDF?

image

File: public/NWS_Hazards_H3/NWS_Hazards_H3.py (hunk 1 above)
Comment (alexlowellmartin) [reply public/NWS_Hazards_H3/NWS_Hazards_H3.py]: Just added the CMAP and an add_rgb_cmap() function to utils.py. Would you recommend I move the GDF geo to cells bit to `utils` as well or look good as is now?

File: public/NWS_Hazards_H3/README.MD (hunk 2 above)
Comment (alexlowellmartin) [reply public/NWS_Hazards_H3/README.MD]: Added to repo

File: public/NWS_Hazards_H3/NWS_Hazards_H3.py (hunk 3 above)
Comment (alexlowellmartin) [reply public/NWS_Hazards_H3/NWS_Hazards_H3.py:15]: Added `bbox: fused.types.TileGDF=None` passed as first parameter to the UDF and added additional params to pass the bbox to the MapServer endpoint.

I'm using `"esriSpatialRelEnvelopeIntersects"` as the spatial relationship - if you'd recommend using `"esriSpatialRelIntersects"` instead please let me know. I didn't notice any material difference in performance between the two.

Comment (pgzmnk) [reply public/NWS_Hazards_H3/NWS_Hazards_H3.py:15]: No opinion here yet.

File: public/NWS_Hazards_H3/NWS_Hazards_H3.py (hunk 1 above)
Comment (pgzmnk) [reply public/NWS_Hazards_H3/NWS_Hazards_H3.py]: It's small enough that it seems ok at the udf level.

File: public/NWS_Hazards_H3/NWS_Hazards_H3.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    
+    params = {
+    "where": "1=1",
+    "outFields": "*", 
+    "f": "geojson",
+    "geometryType": "esriGeometryEnvelope",
+    "geometry": bbox,
```
Comment (pgzmnk) [inline public/NWS_Hazards_H3/NWS_Hazards_H3.py:22]: Setting geometry to `bbox` seems to not have an effect on the query. The parameters are serialized in the request so `bbox` may not be passed as intended. Furthermore, it's unclear from the endpoint documentation how it expects the geometry argument to be specified.

**Pull request merged**